ads_plugin.py

```diff
@@ -0,0 +1,190 @@
+            return
+
+        selectedModel = self.modelCombobox.GetStringSelection()
+
+        if selectedModel == 'other':
+            # Ask the user where the model is located
+            with wx.DirDialog(self, "select the directory in which the model is locatted", defaultPath="",
+                              style=wx.DD_DEFAULT_STYLE | wx.DD_DIR_MUST_EXIST) as fileDialog:
+
+                if fileDialog.ShowModal() == wx.ID_CANCEL:
+                    return
+
+            modelPath = fileDialog.GetPath()
+
+        elif (selectedModel == 'SEM') or (selectedModel == 'TEM'):
+            modelPath = os.path.dirname(AxonDeepSeg.__file__)
+            modelPath = os.path.join(modelPath, 'models', 'default_' + selectedModel + '_model_v1')
+
+        else:
+            print('Please select a model')
+            return
+
+
+        model_configfile = os.path.join(modelPath, 'config_network.json')
+        with open(model_configfile, 'r') as fd:
+            config_network = json.loads(fd.read())
+
+        prediction = axon_segmentation([self.imageDirPath], [self.pngImageName], modelPath,
+                                       config_network, verbosity_level=3)
+
+
+        axonMaskPath = self.imageDirPath + '/AxonDeepSeg_seg-axon.png'
+        myelinMaskPath = self.imageDirPath + '/AxonDeepSeg_seg-myelin.png'
+
+        self.loadPngImageFromPath(axonMaskPath, isMask=True)
+        self.loadPngImageFromPath(myelinMaskPath, isMask=True)
+
+
+    def loadPngImageFromPath(self, imagePath, isMask=False):
+        img_png = np.asarray(Image.open(imagePath).convert('LA'), dtype=np.uint8)
+
+        if np.size(img_png.shape) == 3:
+            img_png2D = img_png[:, :, 0]
+
+
+        elif np.size(img_png.shape) == 2:
+            img_png2D = img_png[:, :]
+
+        else:
+            print("Invalid image dimensions")
+            return
+
+        if isMask is True:
+            img_png2D = img_png2D // 255
+
+        img_NIfTI = nib.Nifti1Image(np.flipud(np.rot90(img_png2D, k=3, axes=(0, 1))), np.eye(4))
+
+        # Save the NIfTI image
+        outFile = imagePath[:-3] + 'nii.gz'
+        nib.save(img_NIfTI, outFile, )
+
+        img_overlay = ovLoad.loadOverlays(paths=[outFile], inmem=True, blocking=True)[0]
+        self.overlayList.append(img_overlay)
+
+    def getCitation(self):
+
+        return ('If you use this work in your research, please cite it as follows: \n'
+                'Zaimi, A., Wabartha, M., Herman, V., Antonsanti, P.-L., Perone, C. S., & Cohen-Adad, J. (2018). '
+                'AxonDeepSeg: automatic axon and myelin segmentation from microscopy data using convolutional '
+                'neural networks. Scientific Reports, 8(1), 3816. '
+                'Link to paper: https://doi.org/10.1038/s41598-018-22181-4. \n'
+                'Copyright (c) 2018 NeuroPoly (Polytechnique Montreal)')
+
+    def getLogo(self):
+        path = os.path.dirname(AxonDeepSeg.__file__)
+
+        logoFile = path + '/ADS_logo.png'
```

Comment:
currently this file does not exist in the repository. Once we create a proper logo (with alpha transparency, etc.), we will likely put it under `https://github.com/neuropoly/axondeepseg/tree/master/docs/source/_static` to replace the current logo.png. So you might as well use that path for now